playground/infrastructure/cloudbuild/examples_cd.sh

```diff
@@ -0,0 +1,161 @@
+ BEAM_ROOT_DIR=$BEAM_ROOT_DIR
+ PROJECT_ID=$PROJECT_ID
+ SDK_CONFIG=$SDK_CONFIG
+ BEAM_EXAMPLE_CATEGORIES=$BEAM_EXAMPLE_CATEGORIES
+ BEAM_CONCURRENCY=$BEAM_CONCURRENCY
+ SDKS=$SDKS
+ COMMIT=$COMMIT
+ DNS_NAME=$DNS_NAME
+ BEAM_USE_WEBGRPC=$BEAM_USE_WEBGRPC
+ NAMESPACE=$NAMESPACE
+ SOURCE_BRANCH=$SOURCE_BRANCH"
+
+# Script starts in a clean environment in Cloud Build. Set minimal required environment variables
+if [ -z "$PATH" ]; then
+    export PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin/:/usr/bin:/sbin:/bin"
+fi
+if [ -z "$HOME" ]; then
+    export HOME="/builder/home"
+fi
+
+export STEP=CD
+
+LogOutput "Installing python and dependencies."
+# Install Python 3 and dependencies
+set -e  # Exit immediately if any command fails
+apt update > /dev/null 2>&1
+export DEBIAN_FRONTEND=noninteractive
+apt install -y apt-transport-https ca-certificates software-properties-common curl unzip apt-utils > /dev/null 2>&1
+add-apt-repository -y ppa:deadsnakes/ppa > /dev/null 2>&1 && apt update > /dev/null 2>&1
+apt install -y python3.8 python3.8-distutils python3-pip > /dev/null 2>&1
+apt install -y --reinstall python3.8-distutils > /dev/null 2>&1
+pip install --upgrade google-api-python-client > /dev/null 2>&1
+python3.8 -m pip install pip --upgrade > /dev/null 2>&1
+ln -s /usr/bin/python3.8 /usr/bin/python > /dev/null 2>&1
+apt install -y python3.8-venv > /dev/null 2>&1
+pip install -r /workspace/beam/playground/infrastructure/requirements.txt > /dev/null 2>&1
+
+LogOutput "Python and dependencies have been successfully installed."
+
+LogOutput "Checking what files were changed in the PR."
+
+git fetch --all > /dev/null 2>&1
+
+diff_log=$(git diff --name-only $DIFF_BASE...forked/$SOURCE_BRANCH)
+diff=($(echo "$diff_log" | tr '\n' ' '))
+LogOutput "List of changed files in PR $SOURCE_BRANCH merging into $DIFF_BASE are:
+$diff_log"
+
+LogOutput "Looking for changes that require CD validation for [$SDKS] SDKs"
+allowlist_array=($ALLOWLIST)
+for sdk in $SDKS
+do
+    eval "example_for_${sdk}_changed"='False'
+    LogOutput "------------------Starting checker.py for SDK_${sdk^^}------------------"
+    cd $BEAM_ROOT_DIR/playground/infrastructure
+    python3 checker.py \
+    --verbose \
+    --sdk SDK_"${sdk^^}" \
+    --allowlist "" \
+    --paths "${diff[@]}"
+    checker_status=$?
+    if [ $checker_status -eq 0 ]; then
+        LogOutput "Checker found changed examples for SDK_${sdk^^}"
+        eval "example_for_${sdk}_changed"='True'
+    elif [ $checker_status -eq 11 ]; then
+        LogOutput "Checker did not find any changed examples for SDK_${sdk^^}"
+        eval "example_for_${sdk}_changed"='False'
+        continue
+    else
+        LogOutput "Error: Checker is broken. Exiting the script."
+        exit 1
+    fi
+
+    result=$(eval echo '$'"example_for_${sdk}_changed")
+
+    if [[ $result == True ]]; then
+        LogOutput "Running ci_cd.py for SDK $sdk"
+
+        export SERVER_ADDRESS=https://${sdk}.${DNS_NAME}
+        python3 ci_cd.py \
```

Comment:
If you are installing custom version of python (3.8) please invoke the desired version directly. As writter here, it's not clear what version of python executable will actually be executed

Reply:
Let's use global Python. No need to create venvs for temp machines